pkg/antctl/raw/multicluster/create/access_token.go

```diff
@@ -0,0 +1,200 @@
+)
+
+type accessTokenOptions struct {
+	namespace      string
+	serviceAccount string
+	roleBinding    string
+}
+
+var accessTokenOpts *accessTokenOptions
+
+var accessTokenExamples = strings.Trim(`
+# Create an access token in a leader cluster for one or more member clusters. If the ServiceAccount or RoleBinding does not exist, the command will create them too
+  $ antctl mc create accesstoken <NAME> -n <NAMESPACE> --service-account <SERVICE_ACCOUNT> --role-binding <ROLE_BINDING>
+`, "\n")
+
+func (o *accessTokenOptions) validateAndComplete() error {
+	if o.namespace == "" {
+		return fmt.Errorf("the Namespace cannot be empty")
+	}
+	if o.serviceAccount == "" {
+		return fmt.Errorf("the ServiceAccount cannot be empty")
+	}
+	if o.roleBinding == "" {
+		return fmt.Errorf("the RoleBinding cannot be empty")
+	}
+	return nil
+}
+
+func NewAccessTokenCmd() *cobra.Command {
+	command := &cobra.Command{
+		Use:     "accesstoken",
+		Args:    cobra.MaximumNArgs(1),
+		Short:   "Create an accesstoken in a leader cluster",
+		Long:    "Create an accesstoken in a leader cluster",
+		Example: accessTokenExamples,
+		RunE:    accessTokenRunE,
+	}
+
+	o := &accessTokenOptions{}
+	accessTokenOpts = o
+	command.Flags().StringVarP(&o.namespace, "namespace", "n", "", "Namespace of the ClusterClaim")
+	command.Flags().StringVarP(&o.serviceAccount, "service-account", "", "", "ServiceAccount of the access token")
+	command.Flags().StringVarP(&o.roleBinding, "role-binding", "", "", "RoleBinding of the ServiceAccount")
+
+	return command
+}
+
+func accessTokenRunE(cmd *cobra.Command, args []string) error {
+	if err := accessTokenOpts.validateAndComplete(); err != nil {
+		return err
+	}
+	if len(args) != 1 {
+		return fmt.Errorf("exactly one NAME is required, got %d", len(args))
+	}
+	kubeconfig, err := raw.ResolveKubeconfig(cmd)
+	if err != nil {
+		return err
+	}
+	restconfigTmpl := rest.CopyConfig(kubeconfig)
+	raw.SetupKubeconfig(restconfigTmpl)
+	k8sClient, err := client.New(kubeconfig, client.Options{Scheme: multiclusterscheme.Scheme})
+	if err != nil {
+		return err
+	}
+
+	serviceAccount := corev1.ServiceAccount{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      accessTokenOpts.serviceAccount,
+			Namespace: accessTokenOpts.namespace,
+		},
+	}
+
+	var createErr error
+	fmt.Fprintf(cmd.OutOrStdout(), "Creating ServiceAccount \"%s\"\n", accessTokenOpts.serviceAccount)
+	createErr = k8sClient.Create(context.TODO(), &serviceAccount)
+	if createErr != nil {
+		if errors.IsAlreadyExists(createErr) {
+			fmt.Fprintf(cmd.OutOrStderr(), "ServiceAccount \"%s\" already exists\n", accessTokenOpts.serviceAccount)
+			createErr = nil
+		} else {
+			fmt.Fprintf(cmd.OutOrStdout(), "Failed to create ServiceAccount \"%s\", error: %s\n", accessTokenOpts.serviceAccount, createErr.Error())
+			return createErr
+		}
+	} else {
+		fmt.Fprintf(cmd.OutOrStdout(), "ServiceAccount \"%s\" created\n", serviceAccount.Name)
+	}
+	defer func() {
+		if createErr != nil {
```

Comment:
This will delete the serviceAccount if it's not created by this command. I think the cleanup should only be defered only when the creation is success and only be executed when returned error is not nil.
```
err = k8sClient.Create(context.TODO(), &serviceAccount)
if err != nil {
    ...
} else {
    defer func() {
        if err != nil {
            ..
        }
    }()
}
```